spec/faraday/rack_builder_spec.rb

```diff
@@ -193,4 +193,149 @@ class Broken < Faraday::Middleware
       end
     end
   end
+
+  context 'when middleware is added with named arguments' do
+    let(:conn) { Faraday::Connection.new {} }
+
+    let(:dog_middleware) do
+      Class.new(Faraday::Middleware) do
+        attr_accessor :name
+        def initialize(app, name:)
+          super(app)
+          @name = name
+        end
+      end
+    end
+    let(:dog) do
+      subject.handlers.find { |handler| handler == dog_middleware }.build
+    end
+
+    it 'adds a handler to construct middleware with options passed to use' do
+      subject.use dog_middleware, name: 'Rex'
+      expect { dog }.to_not output(
+        /warning: Using the last argument as keyword parameters is deprecated/
+      ).to_stderr
+      expect(dog.name).to eq('Rex')
+    end
+  end
+
+  context 'when a request adapter is added with named arguments' do
+    let(:conn) { Faraday::Connection.new {} }
+
+    let(:cat_request) do
+      Class.new(Faraday::Request) do
+        attr_accessor :name
+        def initialize(app, name:)
+          super(app)
+          @name = name
+        end
+      end
+    end
+    let(:cat) do
+      subject.handlers.find { |handler| handler == cat_request }.build
+    end
+
+    it 'adds a handler to construct request adapter with options passed to request' do
+      skip 'still raising last argument kwarg deprecation warnings'
```

Comment:
I think I'm not declaring the Request middleware properly here, because he test still complains about keyword args even when all the methods have the `ruby2_keywords` prefix. I'd appreciate it if you could take a look and let me know how this should be set up for the test!

Reply:
I think the problem here might be the `Faraday::Request` on line 226.
We don't have a specific base class for Request middleware like we have for response ones (`Faraday::Response::Middleware`), so `cat_request` should probably inherit from `Faraday::Middleware` instead.

Reply:
It did work!